chromadb/utils/embedding_functions.py

```diff
@@ -50,6 +50,44 @@ def _verify_sha256(fname: str, expected_sha256: str) -> bool:
     return sha256_hash.hexdigest() == expected_sha256
 
 
+class TogetherAIEmbeddingFunction(EmbeddingFunction[Documents]):
+    # Together AI Embeddings Quick Start Reference
+    # https://docs.together.ai/docs/embeddings-rest
+    # Models List
+    # https://docs.together.ai/docs/embedding-models
+    
+    def __init__(
+        self,
+        api_key: str,
+        model_name: str = "togethercomputer/m2-bert-80M-8k-retrieval",
+    ):
+        self._api_url = "https://api.together.xyz/v1/embeddings"
+        self._model_name = model_name
+        self._session = requests.Session()
+        self._session.headers.update({
+            "Authorization": f"Bearer {api_key}",
+            "Content-Type": "application/json"
+        })
+
+    def __call__(self, input: Documents) -> Embeddings:
+        embeddings = []
+        for text in input:
+            response = self._session.post(
+                self._api_url,
+                json={
+                    "input": text,
+                    "model": self._model_name
+                }
+            )
+
+            if response.status_code == 200:
```

Comment:
If we get another response code this will return `[]` without telling the user what happened. We should raise an error, which links to documentation about what the user can / should do about it.